_posts/2019-12-18-tfhub-0.7.0/tfhub-0.7.0.Rmd

```diff
@@ -0,0 +1,137 @@
+
+## Simple example
+
+The essencial function of tfhub is `layer_hub` which works just like a [keras](https://github.com/rstudio/keras) but allows you to load a complete pre-trained Deep Learning model.
+
+For example you can:
+
+```{r}
+library(tfhub)
+layer_mobilenet <- layer_hub(
+  handle = "https://tfhub.dev/google/tf2-preview/mobilenet_v2/classification/4"
+)
+```
+
+This will download the MobileNet model pre-trained in the ImageNet dataset. tfhub models are cached locally and don't need to be downloaded the next time you use the same model.
+
+You can now use `layer_mobilenet` as an usual Keras layer. For example you can define a model:
+
+```{r}
+library(keras)
+input <- layer_input(shape = c(224, 224, 3))
+output <- layer_mobilenet(input)
+model <- keras_model(input, output)
+summary(model)
+```
+
+```
+Model: "model"
+____________________________________________________________________
+Layer (type)                  Output Shape               Param #    
+====================================================================
+input_2 (InputLayer)          [(None, 224, 224, 3)]      0          
+____________________________________________________________________
+keras_layer_1 (KerasLayer)    (None, 1001)               3540265    
+====================================================================
+Total params: 3,540,265
+Trainable params: 0
+Non-trainable params: 3,540,265
+____________________________________________________________________
+```
+
+This model can now be used to create predict Imagenet labels for an image. For example, let's see the results for the famous Grace Hopper's photo:
+
+![Grace Hopper](images/grace-hopper.jpg)
+
+```{r}
+img <- image_load("images/grace-hopper.jpg", target_size = c(224,224)) %>% 
+  image_to_array()
+img <- img/255
+dim(img) <- c(1, dim(img))
+pred <- predict(model, img)
+imagenet_decode_predictions(pred[,-1,drop=FALSE])
+```
+
+```
+[[1]]
+  class_name class_description    score
+1  n03763968  military_uniform 9.760404
+2  n02817516          bearskin 5.922512
+3  n04350905              suit 5.729345
+4  n03787032       mortarboard 5.400651
+5  n03929855       pickelhaube 5.008665
+```
+
+TensorFlow Hub also offers many other pre-trained image, text and video models.
+All possible models can be found in the TensorFlow hub [website](https://tfhub.dev).
+
+![TensorFlow Hub](images/tfhub.png)
+
+You can find more examples of `layer_hub` usage in the following articles of the TensorFlow for R website:
+
+- [Transfer Learning with tfhub](https://tensorflow.rstudio.com/tutorials/beginners/basic-ml/tutorial_basic_text_classification_with_tfhub/)
+- [Using tfhub with Keras](https://tensorflow.rstudio.com/guide/tfhub/hub-with-keras/)
+- [tfhub Basics](https://tensorflow.rstudio.com/guide/tfhub/intro/)
+- [Text classification example](https://tensorflow.rstudio.com/guide/tfhub/examples/text_classification/)
+
+## Recipes steps
```

Comment:
I would rename this section to `Recipes and Specifications` and remove the `tfdatasets integration` heading, the tfdatsets section feels small enough that we should probably collapse it.